k2/csrc/cuda/debug.h

```diff
@@ -0,0 +1,226 @@
+// k2/csrc/cuda/debug.h
+
+// Copyright (c) 2020, Xiaomi Corporation ( authors: Meixu Song )
+
+// See LICENSE for clarification regarding multiple authors
+
+#ifndef K2_CSRC_CUDA_DEBUG_H_
+#define K2_CSRC_CUDA_DEBUG_H_
+
+/**
+ * Include multiple cuda headers to make host compiler preprocessor happy.
+ *
+ * @todo
+ *  Find a way to avoid this and make .h/.cc with cuda code
+ *  could be parsed by host compiler (specially, GNU-gcc).
+ *  (May assgin to nvcc to take control through change cmake
+ *  compiler and options. Then, it need cmake-3.18 `FindCUDAToolkit`
+ *  or other config helpers to make things easy.)
+ */
+#include <cuda_runtime.h>
+#include <cuda_runtime_api.h>
+#include <device_launch_parameters.h>
+#include <driver_types.h>
+
+#include <stdio.h>
+#include <assert.h>
```

Comment:
Is this header file  from cuda? or from c++ library `<cassert>`? I know there's an `assert` function in cuda header file, but I don't know how `nvcc` decides which one to use.